projects_test.go

```diff
@@ -0,0 +1,51 @@
+package gitlab
+
+import (
+	"testing"
+)
+
+func TestListAllProjects(t *testing.T) {
+	ts, client := Stub("stubs/projects/index.json")
+	defer ts.Close()
+
+	opt := &ListProjectsOptions{}
+	projects, _, err := client.Projects.ListAllProjects(opt)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(projects) != 2 {
+		t.Fail()
+	}
+	if projects[0].Name != "project" {
+		t.Fail()
+	}
+	if projects[1].Name != "project2" {
+		t.Fail()
+	}
+}
+
+func TestGetProject(t *testing.T) {
+	ts, client := Stub("stubs/projects/show.json")
+	defer ts.Close()
+
+	project, _, err := client.Projects.GetProject(1)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if project.Name != "project" {
+		t.Fail()
+	}
+	if project.Namespace.Name != "group" {
+		t.Fail()
+	}
+	if project.Permissions.ProjectAccess.AccessLevel != MasterPermissions {
+		t.Fail()
+	}
+	if project.Permissions.GroupAccess != nil {
+		t.Fail()
+	}
+}
```

Comment:
Very nice! The only thing left is using the a [different func](https://golang.org/pkg/testing/#B.Errorf) to mark the test as failed, so we can pass along some output. Normally this is done in a style/format like this:

```go
t.Errorf("Expected project name %q, got %q", "project", project.Name)
```

The same goes for `t.Fatal(err)` which could be more expressive by typing:

```go
t.Fatalf("Expected error nil, got: %v", err)
```